tests/test_utils.py

```diff
@@ -19,7 +19,9 @@ def constants():
     utils.set_verbose(verbose)
 
 
-@pytest.fixture(scope="module")
+# Ensure that a new test dir is created for every test,
+# since some tests may add/rename files.
+@pytest.fixture(scope="function")
```

Comment:
Yeah, I had the same "eek" reaction.
I know @arounkles knows pytest better, perhaps he has some wisdom to impart here?

Reply:
I just think this fixture was created first when we didn't have any tests that made changes in the directory, so it was fine to share it across tests. But this is definitely safer. Before this PR, the last test was the only one that made changes, so it didn't matter, but I saw unexpected things when I added a test after it. Test order reliance is a BIG NO :) Reminds me of my early days at SNBN :D